EquationServlet.java

```diff
@@ -0,0 +1,196 @@
+            resp.sendError(SC_CONFLICT, "Problem with equation");
+        } catch (NumberFormatException ee) {
+            resp.sendError(SC_CONFLICT, "Problem with variable");
+        } catch (NullPointerException eee) {
+            resp.sendError(SC_CONFLICT, "Not set required parameters");
+        }
+    }
+
+    private List<String> parseIntoPolishNotation(String equation) throws ParseException {
+        if (!(equation.contains("+")
+                || equation.contains("-")
+                || equation.contains("*")
+                || equation.contains("/"))) {
+            throw new ParseException("notValidEquation", 0);
+        }
+        List<String> tokens = new ArrayList<>();
+        StringTokenizer tokenizer = new StringTokenizer(equation.replaceAll(" ", ""), "()+-/*", true);
+        while (tokenizer.hasMoreElements()) {
+            tokens.add(tokenizer.nextToken());
+        }
+        Deque<String> stack = new ArrayDeque<>();
+        List<String> polishNotation = new ArrayList<>();
+        for (String token : tokens) {
+            try {
+                switch (token) {
+                    case "+":
+                    case "-":
+                    case ")":
+                        while ("+-/*".contains(stack.getFirst())) {
+                            polishNotation.add(stack.removeFirst());
+                        }
+                        if (!token.equals(")")) {
+                            stack.addFirst(token);
+                        } else {
+                            if (!stack.getFirst().equals("("))
+                                throw new ParseException("notValidEquation", 0);
+                            stack.removeFirst();
+                        }
+                        break;
+                    case "*":
+                    case "/":
+                        while ("/*".contains(stack.getFirst())) {
+                            polishNotation.add(stack.removeFirst());
+                        }
+                        stack.addFirst(token);
+                        break;
+                    case "(":
+                        stack.addFirst(token);
+                        break;
+                    default:
+                        polishNotation.add(token);
+                }
+            } catch (NoSuchElementException e) {
+                if (!token.equals(")")) {
+                    stack.addFirst(token);
+                }
+            }
+        }
+        while (!stack.isEmpty()) {
+            if (stack.getFirst().equals("("))
+                throw new ParseException("notValidEquation", 0);
+            polishNotation.add(stack.removeFirst());
+        }
+        return polishNotation;
+    }
+
+    private static int calculate(List<String> polishNotation, Map<String, Integer> args) {
+        Deque<Integer> stack = new ArrayDeque<>();
+        for (String token : polishNotation) {
+            int first;
+            int second;
+            switch (token) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A switch statement does not contain a break](https://app.codacy.com/manual/thejerome/IFMO_Servlets_20190915/pullRequest?prid=4723796)